### PR TITLE
Added Java Implementation for Solution 1 of Milk Factory (Bronze)

### PR DESCRIPTION
**Pull request by ChuangSheep**
All test cases passed for the added solution. 

And besides, a question: If I didn't miss anything, shouldn't the time complexity of [Solution 2](https://usaco.guide/problems/usaco-940-milk-factory/solution#solution-2) be O(n^2)? This loop would cost N and DFS will also take N to iterate. So it would be N*N at the end. 
```
for(int i = 0; i < N; i++){
	int ans = g.DFS(i);
	if(ans == N){
		out.println(i + 1); // If the number of visited stations from the DFS is equal to the total number of stations then output.
		used = true;
		break;
	}
}
```

### CODE REVIEW

**Comment by dongliu0426** [comment]
> And besides, a question: If I didn't miss anything, shouldn't the time complexity of [Solution 2](https://usaco.guide/problems/usaco-940-milk-factory/solution#solution-2) be O(n^2)? This loop would cost N and DFS will also take N to iterate. So it would be N*N at the end.
> 
> ```
> for(int i = 0; i < N; i++){
> 	int ans = g.DFS(i);
> 	if(ans == N){
> 		out.println(i + 1); // If the number of visited stations from the DFS is equal to the total number of stations then output.
> 		used = true;
> 		break;
> 	}
> }
> ```

yeah i think it's O(N^2)

File: solutions/usaco-940.mdx [hunk 1]
```diff
@@ -42,6 +42,51 @@ print(ans, file = open('factory.out', 'w'))
 ```
 	
 </PySection>
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class Solver {
```
Comment (maggie-j-liu) [inline solutions/usaco-940.mdx:51]: maybe name the class Factory?

**Review by maggie-j-liu — COMMENTED** [review]
yeah, i think the time complexity of solution 2 is O(N^2)

**Review by SansPapyrus683 — APPROVED** [review]
reviewing on phone so it’s too hard to nitpick
p good tho

**Review by ryanchou-dev — APPROVED** [review]
looks good, happy first pr :)

**Comment by ChuangSheep** [comment]
I did something stupid, fixing...

File: solutions/usaco-940.mdx (hunk 1 above)
Comment (ChuangSheep) [reply solutions/usaco-940.mdx]: Done.

**Comment by ChuangSheep** [comment]
> yeah, i think the time complexity of solution 2 is O(N^2)

Then I'm gonna fix this as well.

**Review by maggie-j-liu — APPROVED** [review]

**Review by dongliu0426 — APPROVED** [review]
yay

**Pull request merged**